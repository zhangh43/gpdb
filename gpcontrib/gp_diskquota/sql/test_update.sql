-- Test Update
CREATE SCHEMA s4;
SELECT diskquota.set_schema_quota('s4', '1 MB');
SET search_path TO s4;
CREATE TABLE a(i int);
INSERT INTO a SELECT generate_series(1,100000000);
SELECT pg_sleep(5);
-- expect update fail.
UPDATE a SET i = 100;
DROP TABLE a;
RESET search_path;
DROP SCHEMA s4;

