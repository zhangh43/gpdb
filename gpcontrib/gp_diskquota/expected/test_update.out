-- Test Update
CREATE SCHEMA s4;
SELECT diskquota.set_schema_quota('s4', '1 MB');
 set_schema_quota 
------------------
 
(1 row)

SET search_path TO s4;
CREATE TABLE a(i int);
NOTICE:  Table doesn't have 'DISTRIBUTED BY' clause -- Using column named 'i' as the Greenplum Database data distribution key for this table.
HINT:  The 'DISTRIBUTED BY' clause determines the distribution of data. Make sure column(s) chosen are the optimal data distribution key to minimize skew.
INSERT INTO a SELECT generate_series(1,100000000);
ERROR:  schema's disk space quota exceeded with name:s4
SELECT pg_sleep(5);
 pg_sleep 
----------
 
(1 row)

-- expect update fail.
UPDATE a SET i = 100;
ERROR:  schema's disk space quota exceeded with name:s4
DROP TABLE a;
RESET search_path;
DROP SCHEMA s4;
